lib/puppet/data_providers.rb

```diff
@@ -25,6 +27,9 @@ def self.lookup_in_module(name, lookup_invocation, merge)
   end
 
   def self.lookup_adapter(lookup_invocation)
+    unless Puppet[:strict] == :off
+      Puppet.deprecation_warning('The method Puppet::DataProviders.lookup_adapter is deprecated and will be removed in the next major release of Puppet.')
```

Comment:
Not likely, since this whole file is now deprecated. The only chance of someone calling here is basically if they'd written something other than data providers (another lookup framework). Puppet no longer calls this method.